### PR TITLE
listing shop items Using OrderDigest

### PR DESCRIPTION
**Pull request by ipdae**
this is testing for new shop

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1141562434100787/1200503122132933) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by ipdae** [comment]
https://s3.console.aws.amazon.com/s3/object/9c-test?region=ap-northeast-2&prefix=9c-order-test.zip

**Comment by hskim881028** [comment]
- Connect the sell action. 
- Print shop purchase and sale list with order digest

File: nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs [hunk 1]
```diff
@@ -144,15 +125,14 @@ private async void AsyncShow(bool ignoreShowAnimation = false)
...
-
         private void Reset()
         {
             ShowNPC();
         }
 
         public void Open()
         {
-            ReactiveShopState.Update();
+            // ReactiveShopState.UpdateBuyDigests();
```
Comment (ipdae) [inline nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs:135]: is it intended?

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 2]
```diff
@@ -311,11 +319,15 @@ private void ResponseCombinationEquipment(ActionBase.ActionEvaluation<Combinatio
         {
             if (eval.Exception is null)
             {
-                var agentAddress = eval.Signer;
+                var agentAddress = States.Instance.AgentState.address;
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:322]: Is this intended? If not, the previous is better.

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs [hunk 3]
```diff
@@ -452,199 +467,142 @@ private void ResponseSell(ActionBase.ActionEvaluation<Sell> eval)
...
+            }
+
+            foreach (var purchaseInfo in purchaseInfos)
+            {
+                if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
                 {
-                    var buyerAvatarAddress = eval.Action.buyerAvatarAddress;
-                    var buyerAvatarStateValue = eval.OutputStates.GetState(buyerAvatarAddress);
-                    if (buyerAvatarStateValue is null)
+                    if (eval.Action.buyerAvatarAddress == avatarAddress) // buyer
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs:557]: This `if` statement is better wrapped around the outer `foreach` statement like below.
```
if (eval.Action.buyerAvatarAddress == avatarAddress) // buyer
{
  foreach (var purchaseInfo in purchaseInfos)
  {
    if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
    {
    }
  }
}
else // seller
{
  foreach (var purchaseInfo in purchaseInfos)
  {
    if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
    {
    }
  }
}
```

File: nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs [hunk 4]
```diff
@@ -153,67 +156,51 @@ private void ResponseBuy(ActionBase.ActionEvaluation<Buy> eval)
...
+            var purchaseInfos = eval.Action.purchaseInfos;
+            foreach (var purchaseInfo in purchaseInfos)
             {
-                var agentAddress = States.Instance.AgentState.address;
-                var purchaseResults = eval.Action.buyerMultipleResult.purchaseResults;
-                foreach (var purchaseResult in purchaseResults)
+                if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
                 {
-                    if (purchaseResult.errorCode == 0)
+                    if (eval.Action.buyerAvatarAddress == avatarAddress) // buyer
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs:173]: This if statement is better wrapped around the outer foreach statement like below.

```
if (eval.Action.buyerAvatarAddress == avatarAddress) // buyer
{
  foreach (var purchaseInfo in purchaseInfos)
  {
    if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
    {
    }
  }
}
else // seller
{
  foreach (var purchaseInfo in purchaseInfos)
  {
    if (!errors.Exists(tuple => tuple.orderId.Equals(purchaseInfo.OrderId)))
    {
    }
  }
}
```

File: nekoyume/Assets/_Scripts/BlockChain/StorePath.cs [hunk 5]
```diff
@@ -18,7 +18,7 @@ public enum Env
         private static readonly IImmutableDictionary<Env, string> DirNames = new Dictionary<Env, string>
         {
             [Env.Production] = "9c",
-            [Env.Development] = "9c_dev",
+            [Env.Development] = "9c-main-partition2",
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/BlockChain/StorePath.cs:21]:
```suggestion
            [Env.Development] = "9c-dev",
```

File: nekoyume/Assets/_Scripts/Helper/Util.cs [hunk 6]
```diff
@@ -45,5 +48,42 @@ public static string GetBlockToTime(int block)
...
+            var state = Game.Game.instance.Agent.GetState(address);
+            if (state is Dictionary dictionary)
+            {
+                return ItemFactory.Deserialize(dictionary);
+            }
+
+            return null;
+        }
+
+        public static ItemBase GetItemBaseByItemId(int itemId)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/Helper/Util.cs:82]:
```suggestion
        public static ItemBase CreateItemBaseByItemId(int itemId)
```

How about you move this method to `ItemFactory`.

File: nekoyume/Assets/_Scripts/UI/Mail.cs [hunk 7]
```diff
@@ -390,5 +403,20 @@ public void TutorialActionClickFirstCombinationMailSubmitButton()
...
+
+        public void Read(BuyerMail buyerMail)
+        {
+            // Deprecated function
+        }
+
+        public void Read(SellerMail sellerMail)
+        {
+            // Deprecated function
+        }
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Mail.cs:420]:
```suggestion
        [Obsolete]
        public void Read(SellCancelMail mail) { }

        [Obsolete]
        public void Read(BuyerMail buyerMail) { }

        [Obsolete]
        public void Read(SellerMail sellerMail) { }
```

File: nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs [hunk 8]
```diff
@@ -74,15 +77,15 @@ public override void Initialize()
         {
             base.Initialize();
 
-            shopItems.SharedModel.SelectedItemView
-                .Subscribe(OnClickShopItem)
+            ObservableExtensions.Subscribe(shopItems.SharedModel.SelectedItemView, OnClickShopItem)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs:80]: Is this intended?

File: nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs [hunk 9]
```diff
@@ -307,56 +275,52 @@ private void OnClickShopItem(ShopItemView view)
...
-            }
-
-            if (result.costume != null)
-            {
-                return result.costume;
-            }
-
-            return (ItemBase)result.tradableFungibleItem;
-        }
+        // public static ItemBase GetItemBase(Buy.PurchaseResult result)
```
Comment (boscohyun) [inline nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs:296]: Is this a necessary comments?

**Review by boscohyun — COMMENTED** [review]
👍

File: nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs (hunk 9 above)
Comment (hskim881028) [reply nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs:296]: It is correct to delete the legacy functions. 👍🏻

File: nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs (hunk 1 above)
Comment (boscohyun) [reply nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs:128]: @hskim881028 ⬆️

**Comment by boscohyun** [comment]
Is this PR still a draft?

File: nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs (hunk 1 above)
Comment (hskim881028) [reply nekoyume/Assets/_Scripts/UI/Shop/ShopBuy.cs:128]: This is an unnecessary comment. I will remove

File: nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs (hunk 4 above)
Comment (hskim881028) [reply nekoyume/Assets/_Scripts/BlockChain/ActionUnrenderHandler.cs]: fixed👍🏻

File: nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs (hunk 3 above)
Comment (hskim881028) [reply nekoyume/Assets/_Scripts/BlockChain/ActionRenderHandler.cs]: fixed👍🏻

**Review by hskim881028 — APPROVED** [review]

**Review by boscohyun — APPROVED** [review]

**Review by Unengine — APPROVED** [review]

**Pull request merged**